batavia/types/Complex.js

```diff
@@ -197,28 +198,151 @@ Complex.prototype.__abs__ = function() {
 /**************************************************
  * Binary operators
  **************************************************/
-
-Complex.prototype.__pow__ = function(exponent) {
+function hyp(x, y) {
+    x = Math.abs(x)
+    y = Math.abs(y)
+    if (x < y) {
+        var temp = x
+        x = y
+        y = temp
+    }
+    if (x === 0) {
+        return 0
+    } else {
+        var yx = y / x
+        return x * Math.sqrt(1 + yx * yx)
+    }
+}
+function quot(a, b) {
+    var r = new Complex(0, 0)
+    var abs_bimag = 0
+    var abs_breal = 0
+    if (b.real < 0) {
+        abs_breal = -b.real
+    } else {
+        abs_breal = b.real
+    }
+    if (b.imag < 0) {
+        abs_bimag = -b.imag
+    } else {
+        abs_bimag = b.imag
+    }
+    if (abs_breal >= abs_bimag) {
+        if (abs_breal === 0) {
+            r = new Complex(0, 0)
+        } else {
+            const ratio = b.imag / b.real
+            const denom = b.real + b.imag * ratio
+            r = new Complex((a.real + a.imag * ratio) / denom, (a.imag - a.real * ratio) / denom)
+        }
+    } else if (abs_bimag >= abs_breal) {
+        const ratio = b.real / b.imag
+        const denom = b.real * ratio + b.imag
+        r = new Complex((a.real * ratio + a.imag) / denom, (a.imag * ratio - a.real) / denom)
+    } else {
+        r = new Complex(NaN, NaN)
+    }
+    return r
+}
+function powu(x, y) {
+    var mask = 1
+    var r = new Complex(1, 0)
+    var p = x
+    while (mask > 0 && y >= mask) {
+        if (y & mask) {
+            r = __mul__(r, p)
+        }
+        mask <<= 1
+        p = __mul__(p, p)
+    }
+    return r
+}
+function powc(x, y) {
+    if (y.real === 0 && y.imag === 0) {
+        return new Complex(1, 0)
+    } else if (x.real === 0 && x.imag === 0) {
+        if (y.imag !== 0 || y.real < 0) {
+            throw new exceptions.ZeroDivisionError.$pyclass(
+                '0.0 to a negative or complex power'
+                )
+        }
+        return new Complex(0, 0)
+    }
+    var vabs = hyp(x.real, x.imag)
+    var l = Math.pow(vabs, y.real)
+    var at = Math.atan2(x.imag, x.real)
+    var phase = at * y.real
+    if (y.imag !== 0) {
+        l /= Math.exp(at * y.imag)
+        phase += y.imag * Math.log(vabs)
+    }
+    var r = l * Math.cos(phase)
+    var im = l * Math.sin(phase)
+    return new Complex(r, im)
+}
+function powi(x, y) {
+    if (x.real === 0 && x.imag === 0) {
+        return new Complex(0, 0)
+    }
+    if (x.real === 0 && (x.imag === 1 || x.imag === -1)) {
+        return powu(x, y)
+    }
+    if (Number(y) >= MAX_INT) {
+        if (Number(y) <= MAX_FLOAT) {
```

Comment:
But I have to consider 2 cases: 1. the number is greater than MAX_INT and smaller than MAX_FLOAT. If this is the case display the message exponentiation error. Otherwise, display the error message: number to big to be converted to float.